### PR TITLE
Add support for Multipart upload to S3 repository integration tests

### PR DESCRIPTION
**Pull request by tlrx**
This commit adds support for Multipart upload to the internal HTTP server used in S3 repository integration tests.

### CODE REVIEW

File: plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java [hunk 1]
```diff
@@ -432,6 +429,18 @@ void executeMultipartUpload(final S3BlobStore blobStore,
         }
     }
 
+    // non-static, package private for testing
+    void assertMultipartUploadSize(final long blobSize) {
```
Comment (original-brownbear) [inline plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java]: We shouldn't call this `assert` when it's not an `assert`. Maybe just call it `ensureMultiPartUploadSize`? :)

**Review by original-brownbear — APPROVED** [review]
LGTM, but I'd apprechiate one renaming before merging :)

Thanks Tanguy!

File: plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java (hunk 1 above)
Comment (tlrx) [reply plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java]: Sure!

**Comment by tlrx** [comment]
Thanks @original-brownbear

**Pull request merged**